src/XMakeCommandLine/XMake.cs

```diff
@@ -226,6 +226,21 @@ private static void HandleConfigurationException(Exception ex)
         }
 
         /// <summary>
+        /// Insert the command executable path as the first element of the args array.
+        /// </summary>
+        /// <param name="args"></param>
+        /// <returns></returns>
+        private static string[] ConstructArrayArg(string[] args)
```

Comment:
I would still re-write this to return a command line string that's equivalent to Environment.CommandLine. Then revert all the #if !FEATURE_GET_COMMANDLINE downstream.